src/main/java/com/FlightPub/Controllers/IndexController.java

```diff
@@ -218,12 +219,12 @@ public String runSearch(@ModelAttribute BasicSearch search, Model model, HttpSes
     @PostMapping("/advancedSearch")
     public String runAdvancedSearch(@ModelAttribute BasicSearch search, Model model, HttpSession session) {
         model = addDateAndTimeToModel(model);
-        List<Flight> flights;
+        List<Flight>[] flights = new ArrayList[2];
         List<StopOver>[] stopOver = new ArrayList[3];
         search.setFlightServices(flightServices);
         search.setLocationServices(locationServices);
         try{
```

Comment:
What is this trying to catch? it might be better coding practice to check for null instead of letting it error?